### ISSUE TITLE
when I use makeIfThenElseCt or MakeConditionalExpression in Java, coredump happened...

### ISSUE THREAD

**Issue by hlee13** [google/or-tools#4040]
**What version of OR-Tools and what language are you using?**
Version: v9.6.2534
Language: Java

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
I use Routing Solver

**What operating system (Linux, Windows, ...) and version?**
Linux 3.10.0

**What did you do?**
```
                    IntVar[] conditions = solver.makeIntVarArray(3, 0, 1);
                    conditions[0] = routing.activeVar(index0);
                    conditions[1] = routing.activeVar(index1);
                    conditions[2] = solver.makeIsEqualVar(routing.vehicleVar(index0), routing.vehicleVar(index1));

                    constraints.add(solver.makeEquality(
                        solver.makeConditionalExpression(
                            solver.makeSumEquality(conditions, 3).var(), 
                            solver.makeDifference(routingDimension.cumulVar(index0), routingDimension.cumulVar(index1)),
                             0)
                        , 0));
```
**What did you expect to see**

**What did you see instead?**

Make sure you include information that can help us debug (full error message, model Proto).

**Anything else we should know about your project / environment**

**Comment by lperron** [comment]
makeSumEquality cannot be converted to an IntVar. The var() call returns a null object.

**Comment by hlee13** [comment]
> makeSumEquality cannot be converted to an IntVar. The var() call returns a null object.

Yes, before you answered, I replaced it with this makeSumGreaterOrEqual. But thank you very much for your kindness

**Comment by lperron** [comment]
Note that  you are creating variables and overwriting them. This is bad.

**Comment by hlee13** [comment]
> Note that you are creating variables and overwriting them. This is bad.

Thank you. Is there such a problem with the function makeSumGreaterOrEqual? 
I am not very familiar with ortoos and the documentation is not very clear

**Comment by lperron** [comment]
No, the MakeIntVarArray fill in the array with integer variables.
Then you overwrite all these.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
